### ISSUE TITLE
installing on macOS

### ISSUE THREAD

**Issue by terfani** [pywr/pywr#1105]
I just tried to install this on macOS with fresh python 3.11.4. I use both conda and pip and I get error:
--using pip
 In file included from pywr/solvers/cython_glpk.c:773:
      In file included from /private/var/folders/35/sbtqhn7x2vl1tk7dd0dgbcsr0000gn/T/pip-build-env-bpvsoa0q/overlay/lib/python3.11/site-packages/numpy/core/include/numpy/arrayobject.h:5:
      In file included from /private/var/folders/35/sbtqhn7x2vl1tk7dd0dgbcsr0000gn/T/pip-build-env-bpvsoa0q/overlay/lib/python3.11/site-packages/numpy/core/include/numpy/ndarrayobject.h:12:
      In file included from /private/var/folders/35/sbtqhn7x2vl1tk7dd0dgbcsr0000gn/T/pip-build-env-bpvsoa0q/overlay/lib/python3.11/site-packages/numpy/core/include/numpy/ndarraytypes.h:1929:
      /private/var/folders/35/sbtqhn7x2vl1tk7dd0dgbcsr0000gn/T/pip-build-env-bpvsoa0q/overlay/lib/python3.11/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: "Using deprecated NumPy API, disable it with "          "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-W#warnings]
      #warning "Using deprecated NumPy API, disable it with " \
       ^
      pywr/solvers/cython_glpk.c:781:10: fatal error: 'glpk.h' file not found
      #include "glpk.h"
               ^~~~~~~~
      1 warning and 1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pywr
Failed to build pywr
ERROR: Could not build wheels for pywr, which is required to install pyproject.toml-based projects

---using conda
it stucks and gives error that it cannot build wheel

**Comment by jetuk** [comment]
We do not produce pre-compiled binaries (wheels) for MacOS on PyPI. However, it should install via conda fine. 

The above error is because Pip is trying to compiled Pywr from source and it is failing to find the GLPK header files. You will need to install these to continue. I'm not familiar enough with Mac to know what is the best way to do that. 

However, I would suggest install via conda would be easiest way unless you plan on modifying Pywr itself. What is the error you have install via conda?

**Comment by terfani** [comment]
thanks. the conda gives the following errors:

**Comment by terfani** [comment]
thanks for the hint on GLPK. I got it setup now. for the benefit of macOS I did the following: 
I installed Homebrew:
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"

I then installed glpk:
brew install glpk

and then tapped into the Homebrew Science repository: 
brew tap brewsci/science

and here I installed lp-solve:
brew install lp_solve

thanks again.
